cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceTaskContextProvider.java

```diff
@@ -111,13 +113,24 @@ protected void shutDown() throws Exception {
    * Returns the {@link BasicMapReduceTaskContext} for the given task.
```

Comment:
I don't think gc is a concern since on shutdown, the JVM is terminating anyways.